File: src/main/java/xyz/lncvrt/galaxyboxpvp/events/SignChangeListener.java
package xyz.lncvrt.galaxyboxpvp.events;

import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvP;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.SignChangeEvent;

public class SignChangeListener implements Listener {
    private final GalaxyBoxPvP plugin;

    public SignChangeListener(GalaxyBoxPvP plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    private void onSignPlace(SignChangeEvent event) {
        if (plugin.isMuted(event.getPlayer())) event.setCancelled(true);
    }
}

File: src/main/kotlin/xyz/lncvrt/galaxyboxpvp/events/SignChangeListener.kt
package xyz.lncvrt.galaxyboxpvp.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.SignChangeEvent
import xyz.lncvrt.galaxyboxpvp.GalaxyBoxPvPPlugin

class SignChangeListener(private val plugin: GalaxyBoxPvPPlugin) : Listener {
    @EventHandler
    private fun onSignPlace(event: SignChangeEvent) {
        if (plugin.isMuted(event.player)) event.isCancelled = true
    }
}